Islandora-CLAW/islandora
# Only adding rdf:type to the first entry in the jsonld graph

**GitHub Issue**: Resolves https://github.com/Islandora-CLAW/CLAW/issues/1112

# What does this Pull Request do?

Only alters the rdf:type of the first entry in the jsonld graph, not all of them.

# How should this be tested?

- Make a repository item and give it a media
- View the media's `?_format=jsonld` route, and notice that _all_ the entries in the RDF graph have an rdf:type of the Media Use, even people and nodes.
- Pull in this PR
- Now you should only see the media have its rdf:type get the extra object, and all the other resources in the graph remain untouched.

e.g. Before:
```
{
   "@graph":[
      {
         "@id":"http:\/\/localhost:8000\/media\/5?_format=jsonld",
         "@type":[
            "http:\/\/pcdm.org\/models#File",
            "http:\/\/pcdm.org\/use#ServiceFile"
         ],
         "http:\/\/purl.org\/dc\/terms\/title":[
            {
               "@value":"2-Service File.jpg",
               "@language":"en"
            }
         ],
         "http:\/\/www.w3.org\/1999\/02\/22-rdf-syntax-ns#label":[
            {
               "@value":"2-Service File.jpg",
               "@language":"en"
            }
         ],
         "http:\/\/schema.org\/author":[
            {
               "@id":"http:\/\/localhost:8000\/user\/1?_format=jsonld"
            }
         ],
         "http:\/\/schema.org\/dateCreated":[
            {
               "@value":"2019-05-09T18:30:20+00:00",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#dateTime"
            }
         ],
         "http:\/\/schema.org\/dateModified":[
            {
               "@value":"2019-05-09T18:30:20+00:00",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#dateTime"
            }
         ],
         "http:\/\/www.ebu.ch\/metadata\/ontologies\/ebucore\/ebucore#height":[
            {
               "@value":"648",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#int"
            }
         ],
         "http:\/\/pcdm.org\/models#fileOf":[
            {
               "@id":"http:\/\/localhost:8000\/node\/2?_format=jsonld"
            }
         ],
         "http:\/\/www.ebu.ch\/metadata\/ontologies\/ebucore\/ebucore#hasMimeType":[
            {
               "@value":"image\/jpeg",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#string"
            }
         ],
         "http:\/\/www.ebu.ch\/metadata\/ontologies\/ebucore\/ebucore#width":[
            {
               "@value":"432",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#int"
            }
         ],
         "http:\/\/schema.org\/sameAs":[
            {
               "@value":"http:\/\/localhost:8000\/sites\/default\/files\/2019-05\/2-Service%20File.jpg"
            }
         ]
      },
      {
         "@id":"http:\/\/localhost:8000\/user\/1?_format=jsonld",
         "@type":[
            "http:\/\/schema.org\/Person",
            "http:\/\/pcdm.org\/use#ServiceFile"
         ]
      },
      {
         "@id":"http:\/\/localhost:8000\/node\/2?_format=jsonld",
         "@type":[
            "http:\/\/pcdm.org\/models#Object",
            "http:\/\/pcdm.org\/use#ServiceFile"
         ]
      }
   ]
}
```

After:
```
{
   "@graph":[
      {
         "@id":"http:\/\/localhost:8000\/media\/5?_format=jsonld",
         "@type":[
            "http:\/\/pcdm.org\/models#File",
            "http:\/\/pcdm.org\/use#ServiceFile"
         ],
         "http:\/\/purl.org\/dc\/terms\/title":[
            {
               "@value":"2-Service File.jpg",
               "@language":"en"
            }
         ],
         "http:\/\/www.w3.org\/1999\/02\/22-rdf-syntax-ns#label":[
            {
               "@value":"2-Service File.jpg",
               "@language":"en"
            }
         ],
         "http:\/\/schema.org\/author":[
            {
               "@id":"http:\/\/localhost:8000\/user\/1?_format=jsonld"
            }
         ],
         "http:\/\/schema.org\/dateCreated":[
            {
               "@value":"2019-05-09T18:30:20+00:00",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#dateTime"
            }
         ],
         "http:\/\/schema.org\/dateModified":[
            {
               "@value":"2019-05-09T18:30:20+00:00",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#dateTime"
            }
         ],
         "http:\/\/www.ebu.ch\/metadata\/ontologies\/ebucore\/ebucore#height":[
            {
               "@value":"648",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#int"
            }
         ],
         "http:\/\/pcdm.org\/models#fileOf":[
            {
               "@id":"http:\/\/localhost:8000\/node\/2?_format=jsonld"
            }
         ],
         "http:\/\/www.ebu.ch\/metadata\/ontologies\/ebucore\/ebucore#hasMimeType":[
            {
               "@value":"image\/jpeg",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#string"
            }
         ],
         "http:\/\/www.ebu.ch\/metadata\/ontologies\/ebucore\/ebucore#width":[
            {
               "@value":"432",
               "@type":"http:\/\/www.w3.org\/2001\/XMLSchema#int"
            }
         ],
         "http:\/\/schema.org\/sameAs":[
            {
               "@value":"http:\/\/localhost:8000\/sites\/default\/files\/2019-05\/2-Service%20File.jpg"
            }
         ]
      },
      {
         "@id":"http:\/\/localhost:8000\/user\/1?_format=jsonld",
         "@type":[
            "http:\/\/schema.org\/Person"
         ]
      },
      {
         "@id":"http:\/\/localhost:8000\/node\/2?_format=jsonld",
         "@type":[
            "http:\/\/pcdm.org\/models#Object"
         ]
      }
   ]
}
```

# Interested parties
@Islandora-CLAW/committers
